src/test/shared/filesystem.test.ts

```diff
@@ -6,271 +6,27 @@
-        })
-
-        it('gets metadata for non-empty file', async () => {
-            await writeFile(filePath, 'Hello, World!', 'utf8')
-            const actual = await stat(filePath)
-
-            assert.ok(actual)
-            assert.strictEqual(actual.isFile(), true)
-            assert.strictEqual(actual.isDirectory(), false)
-        })
-
-        it('rejects if file does not exist', async () => {
-            let error: Error | string | undefined
-
-            try {
-                await stat(filePath)
-            } catch (err) {
-                error = err  as Error | string | undefined
-            } finally {
-                assert.ok(error)
-            }
-        })
-    })
-
-    describe('writeFileAsync', () => {
-        it('writes text data to file', async () => {
-            await writeFile(filePath, 'Hello, World!', 'utf8')
-            const actual = await readFile(filePath, 'utf8')
-
-            assert.ok(actual)
-            assert.strictEqual(actual, 'Hello, World!')
-        })
-
-        it('writes binary data to file', async () => {
-            await writeFile(filePath, 'Hello, World!', 'binary')
-            // tslint:disable-next-line:no-null-keyword
-            const actual = await readFile(filePath)
-
-            assert.ok(actual)
-            assert.strictEqual(actual instanceof Buffer, true)
-            assert.strictEqual((actual as Buffer).toString('binary'), 'Hello, World!')
-        })
-
-        it('overwrites existing file', async () => {
-            await writeFile(filePath, 'Hello, Tom!', 'utf8')
-            await writeFile(filePath, 'Hello, Jane!', 'utf8')
-            const actual = await readFile(filePath, 'utf8')
-
-            assert.ok(actual)
-            assert.strictEqual(actual, 'Hello, Jane!')
-        })
-
-        it('creates file if it doesn\'t already exist', async () => {
-            await writeFile(filePath, 'Hello, World!', 'utf8')
-            const actual = await readFile(filePath, 'utf8')
-
-            assert.ok(actual)
-            assert.strictEqual(actual, 'Hello, World!')
-        })
-
-        it('rejects if directory does not exist', async () => {
-            const wrongFolder = path.join(path.dirname(tempFolder), path.basename(tempFolder) + 'WRONG')
-            const wrongFilePath = path.join(wrongFolder, filename)
-            let error: Error | string | undefined
-
-            try {
-                await writeFile(path.join(tempFolder, wrongFilePath), 'Hello, World!', 'utf8')
-            } catch (err) {
-                error = err  as Error | string | undefined
-            } finally {
-                assert.ok(error)
-            }
+describe('filesystem', () => {
+    functionsToTest.forEach((fxName: string) => {
+        it(`filesystem.${fxName} is same as promisify(fs.${fxName})`, async () => {
+            // @ts-ignore
+            assert.strictEqual(String(promisify(fs[fxName])), String(filesystem[fxName]) // tslint:disable-line
```

Comment:
Please always include the specific warning with `tslint:disable` comments, so that we don't unintentionally ignore other issues. I.e. `// tslint:disable-line:some-rule-name`.

Reply:
✔
Also, upon your suggestion, I experimented for more time than I should have with a number of ways to satisfy the linter and compiler. All of the options I explored made the code it harder to read & understand... acting as a mental detour to the typical code reader... distracting from the meaning and purpose of the code . 

Since both type-assertions and compiler hints have the same effect of telling the compiler to just trust you, we should choose the one which if the best fit for clarity and conciseness for the situation (when considering a bunch of `as` mumbo jumbo). In production code we may choose to be longer winded to enable better compile time type checks and avoid the use of both  `as` other compiler/linter stifling. For tests, we should favor conciseness, coverage and clarity since if the type-assertions/assumptions behind the hints are wrong, the tests will simply fail or error out.